crates/holochain/src/core/workflow/sys_validation_workflow/tests.rs

```diff
@@ -37,31 +37,28 @@ async fn sys_validation_workflow_test() {
 
 #[cfg(feature = "unstable-warrants")]
 #[tokio::test(flavor = "multi_thread")]
-async fn sys_validation_produces_invalid_chain_warrant() {
+async fn sys_validation_produces_invalid_chain_op_warrant() {
+    use crate::test_utils::retry_fn_until_timeout;
+
     holochain_trace::test_run();
     let zome = SweetInlineZomes::new(vec![], 0);
     let (dna, _, _) = SweetDnaFile::unique_from_inline_zomes(zome).await;
 
-    let mut conductors = SweetConductorBatch::from_standard_config(2).await;
-    let ((alice,), (bob,)) = conductors
-        .setup_app("app", [&dna])
-        .await
-        .unwrap()
-        .into_tuples();
-    let alice_pubkey = alice.agent_pubkey().clone();
+    let mut conductor = SweetConductor::from_standard_config().await;
+    let alice = conductor.setup_app("app", [&dna]).await.unwrap();
 
     // - Create an invalid op
-    let mut action = ::fixt::fixt!(CreateLink);
-    action.author = alice_pubkey.clone();
-    let action = Action::CreateLink(action);
```

Comment:
This test kept me busy for many hours, trying to figure out why it kept failing. This is supposed to create a warrant in sys validation. It didn't do that. It created a warrant during app validation.
Futher, the op is modified by being written and read from the database. The assertion a few lines down succeeds, because the op does actually have a `NotStored` record entry. When read and rebuilt from the DHT database after that, the record entry becomes an `NA`, and that is not treated as an invalid op by sys validation.

I chanegd to a `StoreEntry` op that reliably causes a warrant by sys validation.

Reply:
It's these kinds of things that make me think we're not ready to make warrants stable. Even if people wanted to take the risk, if it's that hard for us to test then users have no chance of checking how their code behaves when warrants appear.

Reply:
Yeah, who would've thought that a roundtrip to the database may alter an op in such a way. A lot of the other problems I had I blame on complicated test setup/poor test harnesses.